### ISSUE TITLE
Error with playwright after install

### ISSUE THREAD

**Issue by jmgomezpoveda** [mmabrouk/chatgpt-wrapper#291]
Whenever my session expires I need to run again "chatgpt reinstall". This refreshes the session and asks me to log in again.

Once I quit I see the following message:

```
text-davinci-002-render-sha 1> /quit
GoodBye!
Exception ignored in atexit callback: <bound method ChatGPT._shutdown of <chatgpt_wrapper.backends.browser.chatgpt.ChatGPT object at 0x7fc02aa901f0>>
Traceback (most recent call last):
  File "/home/jose/.local/lib/python3.10/site-packages/chatgpt_wrapper/backends/browser/chatgpt.py", line 152, in _shutdown
    self.cleanup()
  File "/home/jose/.local/lib/python3.10/site-packages/chatgpt_wrapper/backends/browser/chatgpt.py", line 165, in cleanup
    self.browser.close()
  File "/home/jose/.local/lib/python3.10/site-packages/playwright/sync_api/_generated.py", line 13553, in close
    return mapping.from_maybe_impl(self._sync(self._impl_obj.close()))
  File "/home/jose/.local/lib/python3.10/site-packages/playwright/_impl/_sync_base.py", line 104, in _sync
    return task.result()
  File "/home/jose/.local/lib/python3.10/site-packages/playwright/_impl/_browser_context.py", line 448, in close
    await self._channel.send("close")
  File "/home/jose/.local/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 61, in send
    return await self._connection.wrap_api_call(
  File "/home/jose/.local/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 461, in wrap_api_call
    return await cb()
  File "/home/jose/.local/lib/python3.10/site-packages/playwright/_impl/_connection.py", line 96, in inner_send
    result = next(iter(done)).result()
playwright._impl._api_types.Error: Target page, context or browser has been closed
```

That said, after that message chatgpt command-line is perfectly usable, so it seems to be only a cosmetic glitch.

This used to work fine with no errors, but I've recently reinstalled/upgraded chatgpt-wrapper and playwright and this error is new. Sharing in case it means anything to anyone.

**Comment by thehunmonkgroup** [comment]
> Whenever my session expires I need to run again "chatgpt reinstall". This refreshes the session and asks me to log in again.

This seems wrong. You should be able to execute the `/session` command to refresh your session. What you're doing is blowing away the entire Playwright user profile, which is why you have to go through the login process again. `chatgpt reinstall` should really only be used as a last resort if you want to start with a clean slate.

> Once I quit I see the following message:

I"m not super experienced with operating on Playwright contexts, that's legacy code I've inherited.

You can try modifying the `cleanup()` function in `chatgpt_wrapper/backends/browser/chatgpt.py` to also check if the context has any open pages before trying to close the context:

```python
if self.browser and self.browser.pages:
    self.browser.close()
```

Let me know if that helps.

**Comment by jmgomezpoveda** [comment]
I've removed the warning by changing the cleanup function as:

    def cleanup(self):
        self.log.info("Cleaning up")
        if self.page is not None and not self.page.is_closed():
            self.page.close()
        if self.browser and self.browser.pages:
            self.browser.close()
        # remove the user data dir in case this is a second instance
        if self.user_data_dir:
            shutil.rmtree(self.user_data_dir)
        self.play.stop()

**Comment by thehunmonkgroup** [comment]
Are you sure we need this?

```python
    if self.page is not None and not self.page.is_closed():
        self.page.close()
```

I would think `self.browser.close()` would handle closing all pages.

**Comment by jmgomezpoveda** [comment]
Without that part I was still getting the warning.

**Comment by thehunmonkgroup** [comment]
fixed in 0.9.0 release.
